### PR TITLE
getContactByEmailAddress() - iOS and Android - working

### PR DESCRIPTION
**Pull request by robmccoll**
This is nearly a copy and paste of the getContactsByPhoneNumber implementation with the minimum necessary bits swapped out to search by email address.  It has been tested on both platforms.

Edit: drop lame excuse for being lazy because it isn't that much work to add to example / readme.

### CODE REVIEW

File: example/App.js [hunk 1]
```diff
@@ -60,12 +60,17 @@ export default class App extends Component<Props> {
 
   search(text) {
     const phoneNumberRegex = /\b[\+]?[(]?[0-9]{2,6}[)]?[-\s\.]?[-\s\/\.0-9]{3,15}\b/m;
+    const emailAddressRegex = /^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i;
```
Comment (morenoh149) [inline example/App.js]: do we really want to do this? aren't you only supposed to check if there's an `@` in the string

Comment (robmccoll) [reply example/App.js]: It depends. Validating an email address accurately is actually fairly difficult. Email addresses are defined across a number of RFCs and don't really have a single well-defined grammar anywhere. If the goal is just differentiating versus a phone number, checking for @ is probably enough. Differentiating versus a contact name is a little harder. Anyway there are some decently accurate regular expressions here https://emailregex.com/ and a good explanation of the various rules and where they come from.

Comment (morenoh149) [reply example/App.js]: ship it

**Comment by morenoh149** [comment]
will go out in the next release

**Pull request merged**